msg/RadarSector.msg

```diff
@@ -1,4 +1,27 @@
-# A collection of radar scanlines, typically covering a contiguous
+# A collection of marine radar rays, typically covering a contiguous
 # arc of a full circle.
-Header header
-RadarScanline[] scanlines
+
+Header header            # timestamp in the header is the acquisition time of 
+                         # the first ray in the sector.
+                         #
+                         # in frame frame_id, angles are measured around 
+                         # the positive Z axis (counterclockwise, if Z is up)
+                         # with zero angle being forward along the x axis
+
+float32 angle_start      # start angle of this sector [rad]
+float32 angle_increment  # angular distance between measurements [rad]
+
+duration time_increment  # time between rays [seconds] - if your scanner
+                         # is moving, this will be used in interpolating
+                         # position of 3d points
+duration scan_time       # time between two consecutive complete scans 
+                         # or revolutions [seconds]
+
+                         # Warning: On hardware that doesn't provide a timestamp
+                         # time-related fields may be estimated and vary
+                         # from sector to sector as estimates are updated
+
+float32 range_min        # range of first sample in each ray [m]
+float32 range_max        # range of last sample in each ray [m]
+
+RadarEcho[] intensities  # intensity data [device-specific units].
```

Comment:
Is there really no physically-sensible unit these values could be converted to? What are the units called in datasheet? dB, W, something else? I can also imagine converting the values to percentages of the emitted ray (log) power or something similar... Lidars usually report intensity as integers 0-100 or 0-255 (although I don't think this is standardized anywhere).

Reply:
> Is there really no physically-sensible unit these values could be converted to? What are the units called in datasheet? dB, W, something else? I can also imagine converting the values to percentages of the emitted ray (log) power or something similar... Lidars usually report intensity as integers 0-100 or 0-255 (although I don't think this is standardized anywhere).

In the case of the radar I am using, the protocol was reversed-engineered (mostly by the OpenCPN community) so it's not clear what units are used.